pjsip/src/pjsua-lib/pjsua_acc.c

```diff
@@ -3208,28 +3210,30 @@ PJ_DEF(pjsua_acc_id) pjsua_acc_find_for_incoming(pjsip_rx_data *rdata)
         return PJSUA_INVALID_ID;
     }
 
-    uri = rdata->msg_info.to->uri;
+    to_header_uri = rdata->msg_info.to->uri;
+    request_uri = rdata->msg_info.msg->line.req.uri;
 
     PJSUA_LOCK();
 
     /* Use Req URI if To URI is not SIP */
-    if (!PJSIP_URI_SCHEME_IS_SIP(uri) &&
-        !PJSIP_URI_SCHEME_IS_SIPS(uri))
+    if (!PJSIP_URI_SCHEME_IS_SIP(to_header_uri) &&
+        !PJSIP_URI_SCHEME_IS_SIPS(to_header_uri))
     {
         if (rdata->msg_info.msg->type == PJSIP_REQUEST_MSG)
-            uri = rdata->msg_info.msg->line.req.uri;
+            to_header_uri = rdata->msg_info.msg->line.req.uri;
```

Comment:
The named var `to_header_uri` may actually contain request URI instead of To header URI, may it cause confusion later? Why not leave the `to_header_uri` as originally `uri`?

Reply:
I now rewrote even this part. Please take a look at the new commit.

Reply:
You are right. Fixed in the last commit!